vuu-ui/packages/vuu-shell/src/layout-management/useLayoutManager.tsx

```diff
@@ -1,42 +1,41 @@
 import React, { useState, useCallback, useContext, useEffect } from "react";
-import { getLocalEntity, saveLocalEntity } from "@finos/vuu-filters";
-import { LayoutJSON } from "@finos/vuu-layout";
-import { getUniqueId } from "@finos/vuu-utils";
+import { getLocalEntity } from "@finos/vuu-filters";
+import { LayoutJSON, LayoutPersistenceManager } from "@finos/vuu-layout";
 import { LayoutMetadata, Layout } from "./layoutTypes";
 
 export const LayoutManagementContext = React.createContext<{
   layouts: Layout[],
   saveLayout: (n: Omit<LayoutMetadata, "id">) => void
 }>({ layouts: [], saveLayout: () => { } })
 
-export const LayoutManagementProvider = (props: { children: JSX.Element | JSX.Element[] }) => {
-
-  const [layouts, setLayouts] = useState<Layout[]>([])
+export const LayoutManagementProvider = (props: {
+    persistenceManager: LayoutPersistenceManager,
+    children: JSX.Element | JSX.Element[]
+  }) => {
+  const [layouts, setLayouts] = useState<Layout[]>([]);
+  const [tempLayouts, setTempLayouts] = useState<Layout[]>([]);
 
   useEffect(() => {
-    const layouts = getLocalEntity<Layout[]>("layouts")
-    setLayouts(layouts || [])
+    const loadedLayouts = props.persistenceManager.loadLayouts();
+    setLayouts(loadedLayouts || [])
   }, [])
 
   useEffect(() => {
-    saveLocalEntity<Layout[]>("layouts", layouts)
-  }, [layouts])
+    if (tempLayouts.length !== 0) {
+      // Persist layouts
+      const generatedId = props.persistenceManager.saveLayout([...layouts, ...tempLayouts]);
+
+      // Update state
+      const newLayout = tempLayouts[0];
+      newLayout.metadata.id = generatedId;
+      setLayouts(prev => [...prev, newLayout]);
+    }
+  }, [tempLayouts])
 
   const saveLayout = useCallback((metadata: Omit<LayoutMetadata, "id">) => {
-    const json = getLocalEntity<LayoutJSON>("api/vui")
+    const json = getLocalEntity<LayoutJSON>("api/vui");
```

Comment:
This is specific to local-config. Should we be using the persistence manager to get the layout JSON from the appropriate source for us?

Reply:
I think @vferraro-scottlogic had some input on this - two hooks that don't communicate with each other?

Reply:
Answer: refactor the hooks.